tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -305,13 +324,13 @@ private Run runHelper(boolean wantMetadata) {
       }
       idx = 0;
       for (Output<?> o : inputs) {
-        inputOpHandles[idx] = o.getUnsafeNativeHandle();
+        inputOpHandles[idx] = (TF_Operation)o.getUnsafeNativeHandle();
         inputOpIndices[idx] = o.index();
         idx++;
       }
       idx = 0;
       for (Output<?> o : outputs) {
-        outputOpHandles[idx] = o.getUnsafeNativeHandle();
+        outputOpHandles[idx] = (TF_Operation)o.getUnsafeNativeHandle();
```

Comment:
instead of all those explicit casts everywhere, maybe we can add 2 methods to `Output`: `getGraphOperationUnsafeNativeHandle` and `getEagerOperationUnsafeNativeHandle`? I think the code would end up cleaner.

Reply:
Well, it's not that bad. Though, I agree that all those weird methods relying on `Object` references should be totally refactored/removed, but let's leave them like they are for now instead of coming up with workarounds?

Reply:
Ok, we can do this later if it gets really nasty.